### PR TITLE
376 fix disk space metric

### LINKED ISSUE #376: Incorrect calculation of used and free disk space in hardware metrics
It looks like disks that don't belong to Bob are also counted.

Code for disk space metric is here: https://github.com/qoollo/bob/blob/master/bob/src/hw_metrics_collector.rs#L81

### PR DESCRIPTION
**Pull request by Boneyan**
Collector now correctly calculates used disk space
Closes #376

### CODE REVIEW

File: bob/src/hw_metrics_collector.rs [hunk 1]
```diff
@@ -80,32 +83,41 @@ impl HWMetricsCollector {
...
+        let mut total = 0;
+        let mut used = 0;
+        let mut free = 0;
+        
+        for mount_point in disks.keys() {
+            let cm_p = Self::to_cpath(mount_point.as_path());
+            unsafe {
+                let mut stat: statvfs = std::mem::zeroed();
+                if statvfs(cm_p.as_ptr() as *const _, &mut stat) == 0 {
+                    let bsize = Wrapping(stat.f_bsize as u64);
```
Comment (idruzhitskiy) [inline bob/src/hw_metrics_collector.rs:109]: Why do you use wrappings here?

File: bob/src/hw_metrics_collector.rs [hunk 2]
```diff
@@ -80,32 +83,41 @@ impl HWMetricsCollector {
...
-                (total + disk_total, free + disk_free)
-            })
+    fn space(disks: &HashMap<PathBuf, String>) -> (u64, u64, u64) {
+        let mut total = 0;
+        let mut used = 0;
+        let mut free = 0;
+        
+        for mount_point in disks.keys() {
+            let cm_p = Self::to_cpath(mount_point.as_path());
+            unsafe {
```
Comment (idruzhitskiy) [inline bob/src/hw_metrics_collector.rs:106]: Unsafe blocks should be as small as possible, so I think you should create function, wrapping unsafe call and returning `Option<statvfs>`, and work with the result in safe code

File: bob/src/hw_metrics_collector.rs [hunk 3]
```diff
@@ -80,32 +83,41 @@ impl HWMetricsCollector {
...
+                    let bavail = Wrapping(stat.f_bavail as u64);
+                    let bfree = Wrapping(stat.f_bfree as u64);
+                    total += (bsize * blocks).0;
+                    free += (bsize * bavail).0;
+                    used += ((blocks - bfree) * bsize).0;
+                }
+            }
+        }
+
+        (total, free, used)
```
Comment (idruzhitskiy) [inline bob/src/hw_metrics_collector.rs:120]: It's better to use proper struct here, as all members are of the same type, and it's easy to misuse result

**Comment by ikopylov** [comment]
LGTM

**Pull request merged**